lib/millstone.js

```diff
@@ -343,6 +343,58 @@ function readExtension(file, cb) {
     });
 }
 
+// Based on https://github.com/cthackers/adm-zip/blob/f66dc607e0c6f31affa0ff75461537f77296b4ef/util/utils.js#L136
+// to allow writting the contents of the zipfile to a fixed destination, as extractEntryTo doesn't allow
+// to change the file name, only the basename
+function writeFileToAsync (destination, content, overwrite, callback) {
+    fs.exists(destination, function(exists) {
+        if(exists && !overwrite)
+            return callback(false);
+
+        fs.stat(destination, function(err, stat) {
+            if(exists &&stat.isDirectory()) {
+                return callback(false);
+            }
+
+            var folder = path.dirname(destination);
+            fs.exists(folder, function(exists) {
+                if(!exists)
+                    mkdirSync(folder);
+
+                fs.open(destination, 'w', 438, function(err, fd) {
+                    if(err) {
+                        fs.chmod(destination, 438, function() {
```

Comment:
Please, it's tedious but it's mandatory to handle error's when possible.

```js
fs.chmod(destination, 438, function (err) {
    if (err) {
        return callback(err);
    }
    [ ... your code ... ]
});
```

Reply:
Done. I was just copying the implementation from AdmZip (since it's not public) but I've refactored it to follow the conventions.